src/Umbraco.Web/Telemetry/ReportSiteTask.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Umbraco.Core;
+using Umbraco.Core.Configuration;
+using Umbraco.Core.IO;
+using Umbraco.Core.Logging;
+using Umbraco.Web.Scheduling;
+
+namespace Umbraco.Web.Telemetry
+{
+    public class ReportSiteTask : RecurringTaskBase
+    {
+        private IRuntimeState _runtime;
+        private IProfilingLogger _logger;
+        private static HttpClient _httpClient;
+
+        public ReportSiteTask(IBackgroundTaskRunner<RecurringTaskBase> runner, int delayBeforeWeStart, int howOftenWeRepeat, IRuntimeState runtime, IProfilingLogger logger)
+            : base(runner, delayBeforeWeStart, howOftenWeRepeat)
+        {
+            _runtime = runtime;
+            _logger = logger;
+
+            if (_httpClient == null)
+                _httpClient = new HttpClient();
+        }
+
+        /// <summary>
+        /// Runs the background task to send the anynomous ID
+        /// to telemetry service
+        /// </summary>
+        /// <returns>A value indicating whether to repeat the task.</returns>
+        public override async Task<bool> PerformRunAsync(CancellationToken token)
+        {
+            // Try & find file at '/umbraco/telemetrics-id.umb'
+            var telemetricsFilePath = IOHelper.MapPath(SystemFiles.TelemetricsIdentifier);
+
+            if (File.Exists(telemetricsFilePath) == false)
+            {
+                // Some users may have decided to not be tracked by deleting/removing the marker file
+                _logger.Warn<ReportSiteTask>("No telemetry marker file found at '{filePath}' and will not report site to telemetry service", telemetricsFilePath);
+
+                // Stop repeating this task (no need to keep checking)
+                // The only time it will recheck when the site is recycled
+                return false;
+            }
+
+
+            var telemetricsFileContents = string.Empty;
+            try
+            {
+                // Open file & read its contents
+                // It may throw due to file permissions or file locking
+                telemetricsFileContents = File.ReadAllText(telemetricsFilePath);
+            }
+            catch (Exception ex)
+            {
+                // Silently swallow ex - but lets log it (ReadAllText throws a ton of different types of ex)
+                // Hence the use of general exception type
+                _logger.Error<ReportSiteTask>(ex, "Error in reading file contents of telemetry marker file found at '{filePath}'", telemetricsFilePath);
```

Comment:
There is no exit here, the logic will continue below even though the file hasn't been read. It will fail on the GUID parse but better to not rely on logic below when you know you cannot continue?